### PR TITLE
JAVAP-13 The Circle Class exercise implemented

### PR DESCRIPTION
**Pull request by mohammadasheri**
JAVAP-13 The Circle class

### CODE REVIEW

**Comment by mi2shiii** [comment]
Equals and hashcode?

**Comment by mi2shiii** [comment]
Private double radius=1? What does it mean and why?

File: src/main/java/exercise_1/TestCircle.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package exercise_1;
```
Comment (mi2shiii) [inline src/main/java/exercise_1/TestCircle.java]: never use underscore in package definition

File: src/main/java/exercise_1/Circle.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package exercise_1;
+
+public class Circle {
+    private double radius = 1.0;
```
Comment (mi2shiii) [inline src/main/java/exercise_1/Circle.java]: What is this?

**Review by mi2shiii — CHANGES_REQUESTED** [review]

File: src/main/java/exercise_5/TestShape.java [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 public class TestShape {
     public static void main(String[] args) {
+
         Shape s1 = new Circle("red", false, 5.5);
```
Comment (mi2shiii) [inline src/main/java/exercise_5/TestShape.java]: s1 is not a good name, maybe redCircle?

File: .idea/compiler.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mi2shiii) [inline .idea/compiler.xml]: Why have you committed the IDE specific files to the main repo?
put them in .gitigonre file.

File: .idea/compiler.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+<project version="4">
```
Comment (mi2shiii) [inline .idea/compiler.xml]: Why have you committed the IDE specific files to the main repo?
put them in .gitigonre file.

File: .idea/jarRepositories.xml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mi2shiii) [inline .idea/jarRepositories.xml]: Why have you committed the IDE specific files to the main repo?
put them in .gitigonre file.

File: .idea/misc.xml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mi2shiii) [inline .idea/misc.xml]: Why have you committed the IDE specific files to the main repo?
put them in .gitigonre file.

File: .idea/modules.xml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mi2shiii) [inline .idea/modules.xml]: Why have you committed the IDE specific files to the main repo?
put them in .gitigonre file.

File: .idea/uiDesigner.xml [hunk 9]
```diff
@@ -0,0 +1,124 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mi2shiii) [inline .idea/uiDesigner.xml]: Why have you committed the IDE specific files to the main repo?
put them in .gitigonre file.

File: .idea/vcs.xml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mi2shiii) [inline .idea/vcs.xml]: Why have you committed the IDE specific files to the main repo?
put them in .gitigonre file.

File: java-practices.iml [hunk 11]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mi2shiii) [inline java-practices.iml]: Why have you committed the IDE specific files to the main repo?
put them in .gitigonre file.

File: src/main/java/exercise_1/Circle.java [hunk 12]
```diff
@@ -0,0 +1,51 @@
+package exercise_1;
```
Comment (mi2shiii) [inline src/main/java/exercise_1/Circle.java]: never user underscore in package definition.

File: src/main/java/exercise_1/Circle.java [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+
+public class Circle {
+    private double radius;
+
+    public Circle() {
+        radius = 1;
```
Comment (mi2shiii) [inline src/main/java/exercise_1/Circle.java]: What the hell is this?
Why do you have to set a value in constructor? Why don't you simply pass a value instead?

File: src/main/java/exercise_2/Author.java [hunk 14]
```diff
@@ -0,0 +1,53 @@
+package exercise_2;
```
Comment (mi2shiii) [inline src/main/java/exercise_2/Author.java]: same

File: src/main/java/exercise_2/TestAuthor.java [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+    public static void main(String[] args) {
+        Author author = new Author("Mohammad Asheri", "mhm.asheri@gmail.com", 'm');
+        System.out.println(author);
+        author.setEmail("hadi.asheri@gmail.com");
+        System.out.println("Name is: " + author.getName());
+        System.out.println("Eamil is: " + author.getEmail());
+        System.out.println("Gender is: " + author.getGender());
+
+        Book dummyBook = new Book("Java for me", author, 19.95, 99);  // Test Book's Constructor
+        System.out.println(dummyBook);  // Test Book's toString()
```
Comment (mi2shiii) [inline src/main/java/exercise_2/TestAuthor.java]: Don't add comments.

File: src/main/java/exercise_5/Circle.java [hunk 16]
```diff
@@ -0,0 +1,46 @@
+package exercise_5;
+
+public class Circle extends Shape{
+
+    private double radius;
+
+    public Circle() {
+        radius = 1.0;
```
Comment (mi2shiii) [inline src/main/java/exercise_5/Circle.java]: ????

File: src/main/java/exercise_5/Rectangle.java [hunk 17]
```diff
@@ -0,0 +1,60 @@
+package exercise_5;
+
+public class Rectangle extends Shape {
+
+    private double length;
+    private double width;
+
+    public Rectangle() {
+        length = 1.0;
```
Comment (mi2shiii) [inline src/main/java/exercise_5/Rectangle.java]: ???????????????

**Review by mi2shiii — CHANGES_REQUESTED** [review]

File: src/main/java/exercise_5/TestShape.java (hunk 3 above)
Comment (mohammadasheri) [reply src/main/java/exercise_5/TestShape.java]: Yes, you are correct. Code readability is very important. I'll fix it

File: src/main/java/exercise_1/TestCircle.java (hunk 1 above)
Comment (mohammadasheri) [reply src/main/java/exercise_1/TestCircle.java]: Yes sir.

File: src/main/java/exercise_1/Circle.java (hunk 2 above)
Comment (mohammadasheri) [reply src/main/java/exercise_1/Circle.java]: By default, this variable will be initiate with 0.0. But it does not make sense for a circle with a radius of 0. It's better to put this in the constructor.

File: src/main/java/exercise_2/TestAuthor.java (hunk 15 above)
Comment (mohammadasheri) [reply src/main/java/exercise_2/TestAuthor.java]: ok

File: .idea/compiler.xml (hunk 4 above)
Comment (mohammadasheri) [reply .idea/compiler.xml]: Yes, sir.

**Review by mohammadasheri — COMMENTED** [review]
Equals and hashcode added to all required classes.

File: src/main/java/exercise2/TestAuthor.java [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+public class TestAuthor {
+    public static void main(String[] args) {
+        Author author = new Author("Mohammad Asheri", "mhm.asheri@gmail.com", 'm');
+        System.out.println(author);
+        author.setEmail("hadi.asheri@gmail.com");
+        System.out.println("Name is: " + author.getName());
+        System.out.println("Eamil is: " + author.getEmail());
+        System.out.println("Gender is: " + author.getGender());
+
+        Book javaBook = new Book("Java for me", author, 19.95, 99);  // Test Book's Constructor
```
Comment (mi2shiii) [inline src/main/java/exercise2/TestAuthor.java:12]: Don't write comments in the code

**Review by mi2shiii — APPROVED** [review]

**Comment by mi2shiii** [comment]
Interfaces season completed. Some scrum practices applied.

**Pull request merged**